src/proto/message_translation/power_frame_msg.hpp

```diff
@@ -121,21 +123,26 @@ TbotsProto_PowerStatus inline createNanoPbPowerStatus(
  */
 
 TbotsProto_PowerPulseControl inline createNanoPbPowerPulseControl(
-    const TbotsProto::PowerControl& google_control, int kick_slope, int kick_constant,
+    const TbotsProto::PowerControl& google_control, double kick_coeff, int kick_constant,
     int chip_pulse_width)
 {
     TbotsProto_PowerPulseControl nanopb_control =
         TbotsProto_PowerPulseControl_init_default;
 
+    // Safety bounds
+    kick_constant = std::max(kick_constant, MAX_KICK_CONSTANT);
+    kick_coeff    = std::max(kick_coeff, MAX_KICK_COEFFICIENT);
+
     switch (google_control.chicker().chicker_command_case())
     {
         case TbotsProto::PowerControl::ChickerControl::kKickSpeedMPerS:
             nanopb_control.chicker.which_chicker_command =
                 TbotsProto_PowerPulseControl_ChickerControl_kick_pulse_width_tag;
             nanopb_control.chicker.chicker_command.kick_pulse_width =
-                kick_slope *
-                    static_cast<uint32_t>(google_control.chicker().kick_speed_m_per_s()) +
-                kick_constant;
+                static_cast<uint32_t>(
+                    kick_constant * std::exp(kick_coeff * google_control.chicker()
+                                                              .auto_chip_or_kick()
+                                                              .autokick_speed_m_per_s()));
```

Comment:
Is this not `google_control.chicker().kick_speed_m_per_s()` anymore?

Reply:
oops bad copy pasta